src/heuristic.c

```diff
@@ -0,0 +1,221 @@
+ * @param width
+ * 		Amount of pixels in the x-direction.
+ * @return
+ * 		Returns 1 if not sucessful.
+ */
+int generate_img(int width)
+{       
+        //Create image raster file 
+        FILE *fptr;
+        fptr = fopen("../test/testImage.bin", "wb");
+        if(fptr == NULL){
+                return 1;
+        }
+
+        srand(0);
+
+        //Dummy input parameter information
+        input_feature_t input_params;
+        input_params.signed_samples = 1;
+        input_params.dyn_range = 8;
+        input_params.x_size = width;
+        input_params.y_size = Y_PIXELS;
+        input_params.z_size = Z_PIXELS;
+        input_params.in_interleaving = BSQ;
+        input_params.in_interleaving_depth = 1; 
+        input_params.byte_ordering = LITTLE;
+        input_params.regular_input = 1;
+        img_buffer[0] = input_params;
+
+        //Generating random sample band values in specified raster formats
+        if (input_params.in_interleaving == BSQ) {
+                for (int z = 0; z < input_params.z_size; z++) {
+                        for (int y = 0; y < input_params.y_size; y++){
+                                for (int x = 0; x < input_params.x_size; x++) {
+                                        int sample = rand() % (256 + 1 - 0) + 0;
+                                        fprintf(fptr, "%d ", sample);
+                                }
+                                fprintf(fptr, "\n");
+                        }
+                }       
+        } else if (input_params.in_interleaving == BIP || input_params.in_interleaving == BIL) {
+                for (int y = 0; y < input_params.y_size; y++) {
+                        for (int x = 0; x < input_params.x_size * input_params.z_size; x++) {
+                                int sample = rand() % (256 + 1 - 0) + 0;
+                                fprintf(fptr, "%d ", sample);
+                        }
+                        fprintf(fptr, "\n");
+                }                  
+        }
+
+        fclose(fptr);
+}
+
+/**
+ * @brief
+ * 		Calculates the heuristic value of an image and stores it in image header.
+ * @param imgnum
+ * 		Designator for the image.
+ * @return
+ * 		Returns 1 if not sucessful.
+ */
+int heuristic_img(int imgnum)
+{
+        int raw_heuristic = 0;
+        FILE *fptr;
+        fptr = fopen("../test/testImage.bin", "rb");
+
+        if (fptr == NULL) {
+                printf("Unable to open image file");
+                return 1;
+        }
+
+        //Allocate memory for seperate band arrays
+        float *b = malloc(img_buffer[imgnum].x_size * img_buffer[imgnum].y_size * sizeof(float));
+        float *r = malloc(img_buffer[imgnum].x_size * img_buffer[imgnum].y_size * sizeof(float));
+        float *nir = malloc(img_buffer[imgnum].x_size * img_buffer[imgnum].y_size * sizeof(float));
+        float *swir = malloc(img_buffer[imgnum].x_size * img_buffer[imgnum].y_size * sizeof(float));
```

Comment:
May need to include FreeRTOS.h (I think this is what you want) so that you can use the function `pvPortMalloc` -- look it up, I think basically it's just the thread-safe version of malloc from FreeRTOS so that you can, in theory, change contexts in the middle of this function.